### DISCUSSION TITLE
BluetoothA2DPSinkQueued and more Callbacks

### DISCUSSION THREAD

**Discussion by pschatzmann** [pschatzmann/ESP32-A2DP#327] [Announcements]
- Provide BluetoothA2DPSinkQueued class which uses an additional Queue between A2DP and I2S
- Redesign reconnect logic to prevent an automatic reconnect after a disconnect has been requested via the API or from the remote system
- increased volume for bt_music_receiver_32bit_ext example
- Added support for a post I2S init callback (thanks to [adam-puleo](https://github.com/adam-puleo))

<hr /><em>This discussion was created from the release <a href='https://github.com/pschatzmann/ESP32-A2DP/releases/tag/v1.7.2'>BluetoothA2DPSinkQueued and more Callbacks</a>.</em>

**Comment by nonzhe** [comment]
thank you very much. Can you provide an example? I want to temporarily interrupt playing a2dp,and play a short MP3 audio during Bluetooth playback.

**Reply by pschatzmann** [reply]
I suggest that you have a look at the [arduino-audio-tools](https://github.com/pschatzmann/arduino-audio-tools)!
You would basically deactivate the I2S output from this library and take care of it yourself: So you switch feeding it from a decoded file or from a2dp...
